include/revng/Yield/ControlFlow/Graph.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Support/BasicBlockID.h"
+#include "revng/Yield/Support/GraphLayout/Graphs.h"
+
+namespace yield::cfg {
+
+struct Node {
+  BasicBlockID Address;
+  BasicBlockID NextAddress;
+
+  explicit Node(BasicBlockID Address = BasicBlockID::invalid()) :
+    Address(std::move(Address)) {}
+};
+
+enum class EdgeType { Unconditional, Call, Taken, Refused };
+struct Edge {
+  EdgeType Type = EdgeType::Unconditional;
+};
+
+using PreLayoutNode = layout::InputNode<Node, Edge>;
+using PreLayoutGraph = layout::InputGraph<Node, Edge>;
+
+using PostLayoutNode = layout::OutputNode<Node, Edge>;
+using PostLayoutGraph = layout::OutputGraph<Node, Edge>;
```

Comment:
Couldn't all these be the output version and then we do the layouting in place? AFAICT the layout engine work on its own internal graph, so this should be feasible.

Reply:
This looks like a duplicate of [another thread](https://github.com/revng/revng/pull/278#discussion_r1146086216), so I'm closing it as well.